src/main/java/net/mcreator/blockly/java/blocks/SetEventResultBlock.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2020 Pylo and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2020 Pylo and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.blockly.java.blocks;
+
+import net.mcreator.blockly.BlocklyCompileNote;
+import net.mcreator.blockly.BlocklyToCode;
+import net.mcreator.blockly.IBlockGenerator;
+import net.mcreator.blockly.java.BlocklyToProcedure;
+import net.mcreator.generator.template.TemplateGeneratorException;
+import net.mcreator.util.XMLUtil;
+import org.w3c.dom.Element;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SetEventResultBlock implements IBlockGenerator {
+
+	@Override public void generateBlock(BlocklyToCode master, Element block) throws TemplateGeneratorException {
+		if (master instanceof BlocklyToProcedure) {
+			List<Element> elements = XMLUtil.getDirectChildren(block);
+			String value = null;
+			for (Element element : elements) {
+			if (element.getNodeName().equals("field") && element.getAttribute("name").equals("result"))
+				value = element.getTextContent();
+			}
+			// Element value = XMLUtil.getFirstChildrenWithName(block, "result");
+			if (((BlocklyToProcedure) master).getExternalTrigger() != null) {

```

Comment:
Missing cast check and warning compile note if cast failed saying this block is only supported in procedure editor.

Reply:
The cast check is done on line 54, the compile note is on line 78

Reply:
Oops, my bad.